src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -2675,19 +2675,20 @@ public final void applyFloristModifiers() {
   public void applySynergies() {
     int synergetic = this.getRawBitmap(Modifiers.SYNERGETIC);
     if (synergetic == 0) return; // nothing possible
-    Iterator i = Modifiers.synergies.iterator();
+    Iterator<Entry<String, Integer>> i = Modifiers.synergies.entrySet().iterator();
```

Comment:
Why not just change this section to `Modifiers.synergies.forEach((name, mask) -> {`

Reply:
Consistency. Iterators are used in various places in KoLmafia (even when the map/array itself is accessible), so this is just made to match this.

If someone then wants to change those to forEaches, they could just do it for all of them at once.

Reply:
Nah it makes more sense to fix things as we go

Reply:
what do I do if there are *break*s in the loops? I know *continue*s can be replaced with *return*s, but what about those?

Reply:
@gausie this doesn't seem to work. Lambdas require external variables to be final (or effectively final). However, some of these iterations modify variables from outside the loop.

Map.forEach() for when you want to *apply* something to each element. But here, we're looking at each of them, and *tallying* something about the map.

Reply:
(Here, this is because of the line just 4 lines below:
```java
if (delta > 0.0) usefulSynergies |= value;
```

Reply:
Ah ok never mind then. Can we foreach through them instead? I think this iterator typing isn't amazing